types/sse.d.ts

```diff
@@ -0,0 +1,97 @@
+export type SSE = {
+    new (url: string, options?: SSEOptions): SSE;
+    /**
+     * - headers
+     */
+    headers: SSEHeaders;
+    /**
+     * - payload as a string
+     */
+    payload: string;
+    /**
+     * - HTTP Method
+     */
+    method: string;
+    /**
+     * - flag, if credentials needed
+     */
+    withCredentials: boolean;
+    /**
+     * - debugging flag
+     */
+    debug: boolean;
+    FIELD_SEPARATOR: string;
+    listeners: Record<string, Function[]>;
+    xhr: XMLHttpRequest | null;
+    readyState: number;
+    progress: number;
+    chunk: string;
+    addEventListener: AddEventListener;
+    removeEventListener: RemoveEventListener;
+    dispatchEvent: DispatchEvent;
+    stream: Stream;
+    close: Close;
+    onmessage: OnMessage;
+    onopen: OnOpen;
+    onload: OnLoad;
+    onreadystatechange: OnReadystatechange;
+    onerror: OnError;
+    onabort: OnAbort;
+};
+export type SSEHeaders = {
+    [key: string]: string;
+};
+export type SSEOptions = {
+    /**
+     * - headers
+     */
+    headers: SSEHeader;
+    /**
+     * - payload as a string
+     */
+    payload: string;
+    /**
+     * - HTTP Method
+     */
+    method: string;
+    /**
+     * - flag, if credentials needed
+     */
+    withCredentials: boolean;
+    /**
+     * - debugging flag
+     */
+    debug: boolean;
+};
```

Comment:
@mpetazzoni Are these properties all mandatory? If not, you'll want to add `?` behind the property name.

We get this error, because we don't pass `method` and `debug`:

```
error TS2345: Argument of type '{ headers: { "Content-Type": string; }; withCredentials: true; payload: string; }' is not assignable to parameter of type 'SSEOptions'.
  Type '{ headers: { "Content-Type": string; }; withCredentials: true; payload: string; }' is missing the following properties from type 'SSEOptions': method, debug
```

Reply:
Opened a PR to fix these: https://github.com/mpetazzoni/sse.js/pull/43.

Reply:
Yes, these properties should not be required as they all have default values supplied:

https://github.com/mpetazzoni/sse.js/blob/cdb9a03f831006e3d6f4e22ec876671eba31f7af/lib/sse.js#L30-L35